### PR TITLE
Gj  fix wrong coin symbol in wallet explorer

### PR DESCRIPTION
**Pull request by Jibz1**
## Description 

Describe the changes or additions included in this PR.
This PR fixes the wrong coin symbol in some edge case.
https://mysten.atlassian.net/jira/software/c/projects/APPS/boards/8/backlog?view=detail&selectedIssue=APPS-573&issueLimit=100
## Test Plan 

before 
https://explorer.sui.io/transaction/Bm39inPGzCg4Ww8vcZDYy56mLAfpcHzae8PiZLEDKEWE/?network=devnet

after
https://explorer-m0t3ya25q-mysten-labs.vercel.app/transaction/Bm39inPGzCg4Ww8vcZDYy56mLAfpcHzae8PiZLEDKEWE/?network=devnet

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [x ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: apps/explorer/src/utils/getAmount.ts [hunk 1]
```diff
@@ -28,9 +28,18 @@ const getCoinType = (
     const events = txEffects?.events || [];
     const coinType = events
         ?.map((event: SuiEvent) => {
-            const data = Object.values(event).find(
-                (itm) => itm?.owner?.AddressOwner === address
-            );
+            const data = Object.values(event).find((itm) => {
+                if (
+                    itm?.changeType &&
+                    ['Receive', 'Pay'].includes(itm?.changeType) &&
```
Comment (williamrobertson13) [inline apps/explorer/src/utils/getAmount.ts:34]: nit: we shouldn't need optional chaining on `itm.changeType` since we're checking that it is truthy on line 33. Same goes for `itm?.transactionModule`

Comment (williamrobertson13) [inline apps/explorer/src/utils/getAmount.ts:34]: nit: we shouldn't need optional chaining on itm.changeType since we're checking that it is truthy on line 33. Same goes for `itm?.transactionModule`

File: apps/explorer/src/utils/getAmount.ts [hunk 2]
```diff
@@ -28,9 +28,18 @@ const getCoinType = (
...
+                    itm?.transactionModule !== 'gas'
+                ) {
+                    const { owner, sender } = itm;
+                    const { AddressOwner } = owner as { AddressOwner: string };
```
Comment (williamrobertson13) [inline apps/explorer/src/utils/getAmount.ts:38]: is `itm` not typed?

File: apps/explorer/src/utils/getAmount.ts [hunk 3]
```diff
@@ -28,9 +28,18 @@ const getCoinType = (
     const events = txEffects?.events || [];
     const coinType = events
         ?.map((event: SuiEvent) => {
-            const data = Object.values(event).find(
-                (itm) => itm?.owner?.AddressOwner === address
-            );
+            const data = Object.values(event).find((itm) => {
```
Comment (williamrobertson13) [inline apps/explorer/src/utils/getAmount.ts:31]: just for my own understanding, what does this logic actually do? 😄

File: apps/wallet/src/ui/app/helpers/getAmount.ts [hunk 4]
```diff
@@ -23,9 +23,18 @@ const getCoinType = (
     const events = txEffects?.events || [];
     const coinType = events
         ?.map((event: SuiEvent) => {
-            const data = Object.values(event).find(
-                (itm) => itm?.owner?.AddressOwner === address
-            );
+            const data = Object.values(event).find((itm) => {
```
Comment (williamrobertson13) [inline apps/wallet/src/ui/app/helpers/getAmount.ts:26]: same comments as above

**Review by williamrobertson13 — APPROVED** [review]
few small nits but lgtm!
